### ISSUE TITLE
Feature request: allow regex pattern to validate parameters

### ISSUE THREAD

**Issue by LazzariPaolo** [Azure/azure-quickstart-templates#6726]
It would be nice to have the possibility to enforce a certain pattern for the parameters of an ARM template.

**Comment by ggirard07** [comment]
There is a feature request related to this opened in the ARM uservoice
https://feedback.azure.com/forums/281804-azure-resource-manager/suggestions/38546242-arm-template-parameter-validation

**Comment by LazzariPaolo** [comment]
@ggirard07 that link is now dead

**Comment by krishrocks1904** [comment]
any one working on this ?

**Comment by johnwc** [comment]
Yes, this long overdue. AWS CF templates have had this for some time with the `AllowedPattern` field.
